camera.py

```diff
@@ -0,0 +1,56 @@
+class Camera:
+    def __init__(self):
+        #videos_information will contain the cv2 video capture object
+        self.videos_information = None
+        #frame will store the most recent frame of video captured by the video capture object
+        self.frame = None
+        #grabbed is a boolean that stores whether the frame has been grabbed or not
+        self.grabbed = True
+        #read_thread is the thread on which the particular camera is read
+        self.read_thread = None
+        #allows threads to be synchronized if different cameras are used
+        self.read_lock = None
+        #whether the program is still running or not
+        self.running = True
+    #This opens up the camera stream to allow frames to be analyzed through gstreamer
+    def open(self, gstreamer_pipeline_string):
+        try:
+            self.video_capture = cv.VideoCapture(gstreamer_pipeline_string, cv2.CAP_GSTREAMER)
```

Comment:
`cv2` isn't imported.

Reply:
Also, is there a difference between `cv` and `cv2`?

Reply:
Just a version difference, cv2 has more functionality I think. I've never actually used cv. I think it is a very old version.